File: EarlGrey.Common/Weather/Forecast.cs
namespace EarlGrey.Common.Weather {
    public class Forecast {

        /// <summary>Today, Tomorrow, (Day of week)</summary>
        public string Name { get; set; }

        public ForecastTemperature Temperature { get; set; }
        public ForecastTemperature FeelsLikeTemperature { get; set; }
        public WindForecastData Wind { get; set; }
        public double ChanceOfRain { get; set; }
        public DateTime Sunrise { get; set; }
        public DateTime Sunset { get; set; }

        /// <summary>Short text description</summary>
        public string Description { get; set; }

        /// <summary>Icon to use (not the NWS one)</summary>
        public string Icon { get; set; }
    }
}

File: earlgrey.web/src/model/weather/Forecast.ts
import type ForecastTemperature from "./ForecastTemperatiure";
import type WindForecastData from "./WindForecastData";

export default interface Forecast {
    /** Today, Tomorrow, (Day of week) */
    name: string;

    temperature: ForecastTemperature;
    feelsLikeTemperature: ForecastTemperature;
    wind: WindForecastData;

    /** Chance of rain (0–100) */
    chanceOfRain: number;
    uvIndex: number

    /** Sunrise time (ISO 8601 string) */
    sunrise: string;

    /** Sunset time (ISO 8601 string) */
    sunset: string;

    /** Short text description */
    description: string;

    /** Icon to use (not the NWS one) */
    icon: string;
}
